python_modules/libraries/dagster-dg/dagster_dg/completion.py

```diff
@@ -0,0 +1,45 @@
+import click
+import click.shell_completion
+import shellingham
+import typer._completion_shared
+
+from dagster_dg.utils import exit_with_error
+
+
+def install_completion(context: click.Context):
+    # Click on its own does not support automatic installation of completion scripts, but `typer`
+    # does. However, we don't use `typer` for our app, and the typer completion script is not the
+    # same as the click completion script (nor is it compatible with a pure-click app). But the
+    # typer automatic installation routine works for any completion script.
+    #
+    # Therefore, we:
+    #
+    # 1. Generate the click completion script and store it in a string.
+    # 2. Overwrite the typer completion script, stored in an internal dictionary, with the
+    #    click-generated one.
+    # 3. Call the typer installation routine, which will then install the click-generated script.
+    #
+    # This is obviously hacky and non-ideal, but it works and is probably superior to maintaining
```

Comment:
how are we going to pay for this hacky ness in the future? breaking changes between typer and click?